### PR TITLE
[SPARK-10289] [SQL] A direct write API for testing Parquet

### PR DESCRIPTION
**Pull request by liancheng**
This PR introduces a direct write API for testing Parquet. It's a DSL flavored version of the [`writeDirect` method](https://github.com/apache/parquet-mr/blob/apache-parquet-1.8.1/parquet-avro/src/test/java/org/apache/parquet/avro/TestArrayCompatibility.java#L945-L972) comes with parquet-avro testing code. With this API, it's much easier to construct arbitrary Parquet structures. It's especially useful when adding regression tests for various compatibility corner cases.

Sample usage of this API can be found in the new test case added in `ParquetThriftCompatibilitySuite`.

### CODE REVIEW

File: sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetCompatibilityTest.scala [hunk 1]
```diff
@@ -17,11 +17,15 @@
 
 package org.apache.spark.sql.execution.datasources.parquet
 
-import scala.collection.JavaConverters._
+import scala.collection.JavaConverters.{collectionAsScalaIterableConverter, mapAsJavaMapConverter, seqAsJavaListConverter}
```
Comment (marmbrus) [inline sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetCompatibilityTest.scala]: Why the specific imports?

**Comment by marmbrus** [comment]
Seems useful, and only touches test code, so I'm gonna merge into master and 1.5

**Comment by marmbrus** [comment]
Actually does not apply cleanly to branch-1.5, so I'll hold off.

File: sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetCompatibilityTest.scala (hunk 1 above)
Comment (liancheng) [inline sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetCompatibilityTest.scala]: I thought we should be explicit and avoid wildcard imports according to our style guide. But just realized it's OK to have them for implicit methods.

**Comment by liancheng** [comment]
It's OK to not having this merged into branch-1.5. I've resolved SPARK-10289.

**Pull request closed without merging**